filter_plugins/filters.py

```diff
@@ -1,34 +1,25 @@
-import crypt
-import sys
-from random import SystemRandom, shuffle
 import string
+
+from random import SystemRandom
+
 try:
     import passlib.hash
-    from passlib.hash import grub_pbkdf2_sha512
     HAS_PASSLIB = True
-except:
+except Exception as e:
     HAS_PASSLIB = False
 
-def grub2_hash(password, hashtype='sha512', salt=None):
+from ansible import errors
 
-    cryptmethod = {
-        'md5': '1',
-        'blowfish': '2a',
-        'sha256': '5',
-        'sha512': '6',
-    }
 
+def grub2_hash(password, salt=None, iterations=10000):
     if salt is None:
         r = SystemRandom()
         salt = ''.join([r.choice(string.ascii_letters + string.digits) for _ in range(64)])
 
     if not HAS_PASSLIB:
-        if sys.platform.startswith('darwin'):
-            raise errors.AnsibleFilterError('|password_hash requires the passlib python module to generate password hashes on Mac OS X/Darwin')
-        saltstring =  "$%s$%s" % (cryptmethod[hashtype],salt)
-        encrypted = crypt.crypt(password, salt=salt)
+        raise errors.AnsibleFilterError('grub2_hash requires the passlib python module to generate password hashes')
     else:
-        encrypted = grub_pbkdf2_sha512.hash(password)
+        encrypted = passlib.hash.grub_pbkdf2_sha512.hash(password, salt=salt, rounds=iterations)
```

Comment:
I'd recommend using passlib.hash.grub_pbkdf2_sha512.encrypt instead, since it is an alias that will be supported until passlib-2.0 and allows using older versions of passlib than 1.7.  1.6.5 is included in RHEL7, which is the target here.